client/landing/stepper/hooks/use-site-migration.ts

```diff
@@ -0,0 +1,63 @@
+import { FetchStatus } from '@tanstack/react-query';
+import { usePluginAutoInstallation } from './use-plugin-auto-installation';
+import { useSiteMigrationKey } from './use-site-migration-key';
+import { useSiteTransfer } from './use-site-transfer';
+
+const PLUGIN = { name: 'migrate-guru/migrateguru', slug: 'migrate-guru' };
+
+type Status = 'idle' | 'pending' | 'success' | 'error';
+
+const getMigrationKeyStatus = (
+	migrationKey: string | null | undefined,
+	status: FetchStatus,
+	error: Error | null
+): Status => {
+	if ( migrationKey ) {
+		return 'success';
+	}
+	if ( status === 'fetching' ) {
+		return 'pending';
+	}
+
+	if ( error ) {
+		return 'error';
+	}
+
+	return status as Status;
+};
+
+/**
+ *  Hook to manage the site migration flow, including site transfer, plugin installation and migration key retrieval
+ */
+export const useSiteMigration = ( siteId?: number ) => {
```

Comment:
naming is confusing. This is not the "site migration" hook it's the "prepare site to be migrated into" hook. Maybe `usePrepareSiteForMigration` is better?

Reply:
I liked the suggestion so I renamed here 565136813f156803d48fe5ef1dda2b462bcdf7f

Fortunately, hook names are easy to rename. 🎉